velox/functions/prestosql/fuzzer/WindowFuzzerTest.cpp

```diff
@@ -0,0 +1,151 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/init/Init.h>
+#include <gflags/gflags.h>
+#include <gtest/gtest.h>
+#include <unordered_set>
+
+#include "velox/exec/fuzzer/DuckQueryRunner.h"
+#include "velox/exec/fuzzer/WindowFuzzerRunner.h"
+#include "velox/functions/prestosql/aggregates/RegisterAggregateFunctions.h"
+#include "velox/functions/prestosql/fuzzer/ApproxDistinctInputGenerator.h"
+#include "velox/functions/prestosql/fuzzer/ApproxPercentileInputGenerator.h"
+#include "velox/functions/prestosql/fuzzer/MinMaxInputGenerator.h"
+#include "velox/functions/prestosql/window/WindowFunctionsRegistration.h"
+
+DEFINE_int64(
+    seed,
+    0,
+    "Initial seed for random number generator used to reproduce previous "
+    "results (0 means start with random seed).");
+
+DEFINE_string(
+    only,
+    "",
+    "If specified, Fuzzer will only choose functions from "
+    "this comma separated list of function names "
+    "(e.g: --only \"min\" or --only \"sum,avg\").");
+
+namespace facebook::velox::exec::test {
+namespace {
+
+std::unordered_map<std::string, std::shared_ptr<InputGenerator>>
+getCustomInputGenerators() {
+  return {
+      {"min", std::make_shared<MinMaxInputGenerator>("min")},
+      {"min_by", std::make_shared<MinMaxInputGenerator>("min_by")},
+      {"max", std::make_shared<MinMaxInputGenerator>("max")},
+      {"max_by", std::make_shared<MinMaxInputGenerator>("max_by")},
+      {"approx_distinct", std::make_shared<ApproxDistinctInputGenerator>()},
+      {"approx_set", std::make_shared<ApproxDistinctInputGenerator>()},
+      {"approx_percentile", std::make_shared<ApproxPercentileInputGenerator>()},
+  };
+}
+
+} // namespace
+} // namespace facebook::velox::exec::test
+
+int main(int argc, char** argv) {
+  facebook::velox::aggregate::prestosql::registerAllAggregateFunctions(
+      "", false);
+  facebook::velox::window::prestosql::registerAllWindowFunctions();
+
+  ::testing::InitGoogleTest(&argc, argv);
+
+  // Calls common init functions in the necessary order, initializing
+  // singletons, installing proper signal handlers for better debugging
+  // experience, and initialize glog and gflags.
+  folly::Init init(&argc, &argv);
+
+  size_t initialSeed = FLAGS_seed == 0 ? std::time(nullptr) : FLAGS_seed;
+
+  auto duckQueryRunner =
+      std::make_unique<facebook::velox::exec::test::DuckQueryRunner>();
+  duckQueryRunner->disableAggregateFunctions({
+      "skewness",
+      // DuckDB results on constant inputs are incorrect. Should be NaN,
+      // but DuckDB returns some random value.
+      "kurtosis",
+      "entropy",
+  });
+
+  // List of functions that have known bugs that cause crashes or failures.
+  static const std::unordered_set<std::string> skipFunctions = {
+      // https://github.com/facebookincubator/velox/issues/3493
```

Comment:
@kagamiori I noticed that this issue is closed. Should we remove stddev_pop from the skip list?